src/org/sosy_lab/java_smt/basicimpl/IndependentInterpolatingProverEnvironment.java

```diff
@@ -127,16 +127,34 @@ public BooleanFormula getInterpolant(Collection<TFormulaInfo> pFormulasOfA)
     BooleanFormula conjugatedA = bmgr.and(formulasOfA);
     BooleanFormula conjugatedB = bmgr.and(formulasOfB);
 
-    // handle empty interpolation groups and trivial interpolants
+    ProverEnvironment prover = getDistinctProver();
+    prover.push(conjugatedA);
```

Comment:
Why are you checking SAT here?
We already know that the current state of `A AND B`  has to be UNSAT.
Either there are trivial interpolants that can't be covered by the implemented procedures, or the procedures will produce an interpolant.
And trivial interpolants can be found without ever invoking a SAT check.